### PR TITLE
Replace printing with tracing

### LINKED ISSUE #8: Unsolicited IO from library code
I upgraded to `1.2.1` and it fixed this problem. Now I upgraded to `1.2.4` and the problem is back, albeit in different places, but there're still spurious print statements on errors.

These should either be log statements (ideally with `tracing`) or not be present at all, since this is a library being called by apps with different expectations about IO.

I'll be happy to do the work of converting them to `tracing` calls if you're OK with it.

### PR DESCRIPTION
**Pull request by xandkar**
Fixes #8

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "nom-exif"
-version = "1.2.6"
+version = "1.3.0"
```
Comment (xandkar) [inline Cargo.toml:3]: I chose to bump minor since this is kind of a new feature, but I don't feel super-strongly about that.

**Comment by xandkar** [comment]
In case you decide to make tracing an optional feature, I did it in a separate branch: https://github.com/xandkar/nom-exif/commit/e94d9dc9c3f1e0f5ed2310b07330b96348378409

It is probably going to be too annoying to work with at this stage, hence not part of this PR :)

**Comment by xandkar** [comment]
And there's always the possibility to statically disable tracing levels: https://docs.rs/tracing/latest/tracing/level_filters/index.html#compile-time-filters

**Review by mindeng — APPROVED** [review]
Awesome! Great job!

**Comment by mindeng** [comment]
I am currently on vacation and may not be able to release a new version for the time being. I will try my best to find an opportunity to release it. Thank you for your brilliant work!

**Comment by xandkar** [comment]
Thank you, Min!

**Comment by mindeng** [comment]
@xandkar Version 1.3.0 has been released!

**Pull request merged**